### PR TITLE
Frequency Graph User Data

### LINKED ISSUE #239: DSNP Implementation Update: Graph
Update the Frequency Implementation to support the new Graph and User Data parts of #233

### PR DESCRIPTION
**Pull request by wilwade**
Problem
=======
Update the Frequency Implementation post #233 and #226 

Closes #239

Solution
========
- Added Frequency/UserData.md
- Added PublicKey Announcement
- Added information about the Stateful Storage Pallet
- Updated other related tables.

### CODE REVIEW

File: pages/Frequency/Operations.md [hunk 1]
```diff
@@ -7,19 +7,22 @@ A Pallet is a Substrate runtime module and also a Cargo crate.
 ### MSA Pallet
 Responsible for DSNP Identity Operations and Delegation management.
 
+### Schema Pallet
+Responsible for managing the data structures for DSNP Announcements and User Data.
+
 ### Messages Pallet
-Responsible for DSNP Announcement Operations.
+Responsible for most DSNP Announcement Operations.
```
Comment (shannonwells) [inline pages/Frequency/Operations.md:14]: might be good to list the exceptions or link to them

**Review by shannonwells — APPROVED** [review]
Looks good, I just had a tiny suggestion that is non-blocking.

File: pages/DSNP/UserData.md [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+Data may be public or stored in encrypted form.
+The [Avro](https://avro.apache.org) schema specification is used to define the binary representation of relevant data types.
+
+## User Data Types
+
+DSNP implementations MUST support the following user data types:
+
+| System Name | Version | Encryption Algorithm | Compression Codec | Avro Object Type |
+| --- | --- | --- | --- | --- |
+| `publicFollows` | 1.2 | NONE | [`brotli`](https://brotli.org) | [GraphEdge](Graph.md#edges) |
```
Comment (aramikm) [inline pages/DSNP/UserData.md:18]: I  thought we decided to go with Deflate?

Comment (wilwade) [reply pages/DSNP/UserData.md:18]: I just haven't rebased yet.

**Review by aramikm — CHANGES_REQUESTED** [review]
After looking into the details of how libsodium library and how it needs to get build on each platform and architecture differently and considering that we would need to create bridges for our Rust library on top of this, then it's self evident that how complicated this might become.
I'm recommending to use a Rust native implementation of libsodium. `crypto_box` crate has the same `seal` and `unseal` functionality which is compatible with libsodium.
Other functionalities used in libsodium like `crypto_box_beforenm` and `crypto_kdf_derive_from_key`, `crypto_secretbox_detached` are kinda very niche use-cases and not directly implemented in it but we can reach the same result (more or less) with some other Rust libraries.

In general I'm suggesting to use more mainstream functions and definitions for above functionalities if possible.

File: pages/Frequency/Publishing.md [hunk 3]
```diff
@@ -1,20 +1,22 @@
...
+| 5 | [Profile](../DSNP/Types/Profile.md) | Batched | TBD | 5 | [`Parquet`](https://libertydsnp.github.io/frequency/common_primitives/schema/enum.ModelType.html#variant.Parquet) | [`IPFS`](https://libertydsnp.github.io/frequency/common_primitives/schema/enum.PayloadLocation.html#variant.IPFS) |
+| 6 | [Update](../DSNP/Types/Update.md) | Batched | TBD | 6 | [`Parquet`](https://libertydsnp.github.io/frequency/common_primitives/schema/enum.ModelType.html#variant.Parquet) | [`IPFS`](https://libertydsnp.github.io/frequency/common_primitives/schema/enum.PayloadLocation.html#variant.IPFS) |
+| 7 | [Public Key](../DSNP/Types/PublicKey.md) | [Stateful](./UserData.md#announcements) | TBD | 7 | [`AvroBinary`](https://libertydsnp.github.io/frequency/common_primitives/schema/enum.ModelType.html#variant.AvroBinary) | [`Itemized`](https://libertydsnp.github.io/frequency/common_primitives/schema/enum.PayloadLocation.html#variant.Itemized) |
```
Comment (aramikm) [inline pages/Frequency/Publishing.md:17]: I think the question is that, are these public keys general purpose or only for graph at this point? This stems from the conversion needed between different  X25519 keys into Ed25519 keys to allow signature using the same keys.

File: pages/Frequency/Records.md [hunk 4]
```diff
@@ -45,3 +45,4 @@ State data is retrieved via state queries (`pallet.stateQuery`) or RPC calls (`p
...
 | <a id="failure">Failure Record</a> | See section on [Failure Handling](./Operations.md#failure-handling) |
+| <a id="user-data-replaced">User Data Replaced Record</a> | Event: [`stateful_storage::ItemizedPageUpdated`](https://libertydsnp.github.io/frequency/pallet_stateful_storage/pallet/enum.Event.html#variant.ItemizedPageUpdated), [`stateful_storage::ItemizedPageDeleted`](https://libertydsnp.github.io/frequency/pallet_stateful_storage/pallet/enum.Event.html#variant.ItemizedPageDeleted), [`stateful_storage::PaginatedPageUpdated`](https://libertydsnp.github.io/frequency/pallet_stateful_storage/pallet/enum.Event.html#variant.PaginatedPageUpdated), [`stateful_storage::PaginatedPageDeleted`](https://libertydsnp.github.io/frequency/pallet_stateful_storage/pallet/enum.Event.html#variant.PaginatedPageDeleted)<br />See Also: [Read User Data](./UserData.md#read-operation-mapping) |
```
Comment (aramikm) [inline pages/Frequency/Records.md:48]: It seems like for the other rows we included Rpcs but not for this one.

**Review by aramikm — APPROVED** [review]
Looks good! I added a couple of questions and comments.

**Pull request merged**